mockserver/data_reader.go

```diff
@@ -28,12 +28,73 @@ type BlockHeader struct {
 	NextBlock chainhash.Hash `json:"nextblockhash"`
 }
 
+type Transaction struct {
+	TxId          chainhash.Hash `json:"txid"`
+	Hash          chainhash.Hash `json:"hash"`
+	Version       int32          `json:"version"`
+	Size          int32          `json:"size"`
+	VSize         int32          `json:"vsize"`
+	Weight        int32          `json:"weight"`
+	LockTime      uint32         `json:"locktime"`
+	VIn           []Vin          `json:"vin"`
+	VOut          []Vout         `json:"vout"`
+	Hex           string         `json:"hex"`
+	BlockHash     chainhash.Hash `json:"blockhash"`
+	Confirmations uint32         `json:"confirmations"`
+	Time          uint32         `json:"time"`
+	BlockTime     uint32         `json:"blocktime"`
+}
+
+// Vout models parts of the tx data.  It is defined separately since both
+// getrawtransaction and decoderawtransaction use the same structure.
+type Vout struct {
+	Value        float64            `json:"value"`
+	N            uint32             `json:"n"`
+	ScriptPubKey ScriptPubKeyResult `json:"scriptPubKey"`
+}
+
+// copied from btcjson/chainsvrresults.go and
+// modified to be compatible with bitcoin v1
```

Comment:
Why we need make it compatible with btc v1. What is btc v1 in this context?

Reply:
I meant bitcoin core version that was around initial blocks, `v1` was misleading here.